### PR TITLE
Add alifestd_add_global_root function to auxiliary library

### PR DESCRIPTION
**Pull request by mmore500**
## Summary
This PR adds a new utility function `alifestd_add_global_root` to the auxiliary library for manipulating phylogeny dataframes in alife standard format. The function adds a new global root node that all existing root nodes point to, enabling conversion of multi-root phylogenies into single-root phylogenies.

## Key Changes
- **New function `alifestd_add_global_root`**: Adds a synthetic global root node to a phylogeny dataframe
  - Creates a new root node with ID one greater than the maximum existing ID
  - Updates all existing root nodes to point to the new global root via `ancestor_id` and/or `ancestor_list` columns
  - Supports optional `origin_time` parameter for setting the new root's origin time
  - Includes `mutate` parameter to control whether the input dataframe is modified
  - Only populates applicable columns (`id`, `ancestor_id`, `ancestor_list`, `origin_time`) in the new root row; other columns are NaN

- **Comprehensive test suite** with 30+ test cases covering:
  - Empty dataframes
  - Single and multiple root scenarios
  - Both `ancestor_id` and `ancestor_list` column formats
  - `origin_time` parameter handling
  - Non-contiguous ID values
  - Mutation behavior (mutate=True/False)
  - Warning generation for columns that may become invalid (branch_length, edge_length, node_depth, etc.)
  - Edge cases like single-node phylogenies

## Implementation Details
- The function warns users when columns that may become invalid after adding a global root are present (e.g., `branch_length`, `edge_length`, `node_depth`, `num_descendants`, etc.)
- Uses `alifestd_find_root_ids` to identify existing root nodes
- Leverages pandas operations for efficient dataframe manipulation
- Maintains dataframe immutability by default (only mutates when explicitly requested)
- Validates output using `alifestd_validate` in tests

https://claude.ai/code/session_01AXbrSTa2EbkaaZo7RibWj9

### CODE REVIEW

File: hstrat/_auxiliary_lib/_alifestd_add_global_root.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    if "ancestor_id" in phylogeny_df:
+        phylogeny_df.loc[root_mask, "ancestor_id"] = new_root_id
+
+    if "ancestor_list" in phylogeny_df:
+        phylogeny_df.loc[root_mask, "ancestor_list"] = f"[{new_root_id}]"
+
+    # Append the new root row
+    new_root_df = pd.DataFrame([new_root])
+    res = pd.concat([phylogeny_df, new_root_df], ignore_index=True)
```
Comment (mmore500) [inline hstrat/_auxiliary_lib/_alifestd_add_global_root.py]: how does this handle when the columns don't match? is a kwarg necessary for that (i.e., to silence warnings for mismatched columns; also ensure that the concat is vertical (not horizontal)

File: tests/test_hstrat/test_auxiliary_lib/test_alifestd_add_global_root.py
Comment (mmore500) [inline tests/test_hstrat/test_auxiliary_lib/test_alifestd_add_global_root.py]: Also test where only ancestor_id is provided, and where ancestor_list is sexual (i.e., more than one ancestor for some) --- there might be some edge cases there on the second one especially

**Review by mmore500 — COMMENTED** [review]
changes requested

**Review by mmore500 — COMMENTED** [review]
changes requested

File: hstrat/_auxiliary_lib/_alifestd_add_global_root.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            f"root_attrs must not contain reserved keys {bad_keys}; "
+            "these are set automatically"
+        )
+
+    if not mutate:
+        phylogeny_df = phylogeny_df.copy()
+
+    # Create new root id
+    new_root_id = (
+        phylogeny_df["id"].max() + 1 if len(phylogeny_df) else 0
```
Comment (mmore500) [inline hstrat/_auxiliary_lib/_alifestd_add_global_root.py]: Use .empty

**Review by mmore500 — COMMENTED** [review]
Changes requested; also, add and test a CLI

**Review by mmore500 — COMMENTED** [review]
lgtm; ready to merge once CI passes
